documentation/docs/30-advanced/60-images.md

```diff
@@ -133,13 +133,80 @@ By default, enhanced images will be transformed to more efficient formats. Howev
+```svelte
+<script>
+	import { getImage } from '$app/images';
+</script>
+
+<img {...getImage({ src: '/path/to/image', width: 1000, height: 800 })} alt="An alt text" />
+```
+
+The result in this case will be generated `src` and `srcset` attributes. The `srcset` will contain two resolution variants and the browser decides based on the device's pixel density which image to load.
+
+If you provide the `size` property or set `layout: "fill"` instead of `width` and `height`, the `srcset` will instead contain a list of URLs for various device widths and the browser will choose the appropriate one.
+
+```svelte
+<script>
+	import { getImage } from '$app/images';
+</script>
+
+<img {...getImage({ src: '/path/to/image', width: 1000, height: 800, size: '100vw' })} alt="An alt text" />
+```
+
+Remember to give the largest image above the fold (i.e. the largest one that is immediately visible) a `fetchpriority` of `high` to ensure browsers download it earlier and set `loading: 'lazy'` for unimportant images, which both contributes to better web vital scores.
+
+```svelte
+<script>
+	import { getImage } from '$app/images';
+</script>
+
+<img {...getImage({ src: '/path/to/image', width: 1000, height: 800, size: '100vw', fetchpriority: 'high' })} alt="An alt text" />
+```
+
+Because `getImage` is just a simple function, you can wrap it in a function to set sensible defaults for your app, or use it as a base for a custom image component.
+
+### Configurating an image loader
+
+`getImage` calculates the variants for an image based on its input and the `kit.images.width` configuration, which contains an array of numbers corresponding to various screen widths. For each variant, the configured image loader is invoked when running in production - at dev time, the loader is not invoked to prevent accidental image CDN costs.
+
+By default, no loader is configured and as such the image url is handed back without any modifications. To configure an image loader, pass a string to `kit.images.loader` which represents a path to a file on disk (either a relative path, or referencing an npm package). Some adapters like [`adapter-vercel`](adapter-vercel) provide ready-made loaders. It's also very easy to create your own loader. Here's an example using [`unpic`](https://unpic.pics/lib/), which provides solutions for a wide range of image CDNs:
+
+```js
+/// file: custom-loader.js
+// @filename: ambient.d.ts
+declare module 'unpic' {
+	export function transformUrl(args: any): string;
```

Comment:
hmm. this shouldn't be necessary. `unpic` ships `.d.ts` files

Reply:
Yes but shiki doesn't know what `unpic` is, because it's not any dependency of ours here (and I don't want to add it as a dev dependency just so that we don't need it), therefore I have to mock it.